src/IndexService/Worker/AbstractWorker.php

```diff
@@ -135,7 +135,18 @@ abstract protected function doDeleteFromIndex(int $subObjectId, IndexableInterfa
     protected function convertArray(array|string $data): array|string
     {
         if (is_array($data)) {
-            return WorkerInterface::MULTISELECT_DELIMITER . implode(WorkerInterface::MULTISELECT_DELIMITER, $data) . WorkerInterface::MULTISELECT_DELIMITER;
+            // One-dimensional array
+            if(count($data) == count($data, COUNT_RECURSIVE)) {
+                return WorkerInterface::MULTISELECT_DELIMITER . implode(WorkerInterface::MULTISELECT_DELIMITER, $data) . WorkerInterface::MULTISELECT_DELIMITER;
+            }
+
+            // Two-dimensional array
+            $string = WorkerInterface::MULTISELECT_DELIMITER;
+            foreach($data as $d) {
+                $string .= implode(WorkerInterface::MULTISELECT_DELIMITER, $d);
```

Comment:
this is not working as intended. 

```php 
$data = [[1.1, 1.2], [2.1, 2.2]];
```

results in 

```
"#;#1.1#;#1.22.1#;#2.2#;#"
```

so the delimiter between the two sub arrays is missing.

Reply:
on second though, I think that the better place to fix this might be in `DefaultBrickGetterSequenceToMultiselect` itself ... and make sure it returns already one dimensional arrays, WDYT?

Reply:
@fashxp thank you for the review & suggestions! I've reverted changes from AbstractWorker and adapted the DefaultBrickGetterSequenceToMultiselect. Hopefully it will work well now.